text/0052-resource-importing-support.md

```diff
@@ -0,0 +1,365 @@
+  to perform the import (some manual template updates are needed though) and
+* providing this feature outside of main CDK CLI - e.g. as a separate Node.JS
+  package.
+
+### What are the drawbacks of this solution?
+
+This solution is only a CDK layer on top of CloudFormation resource import
+feature and so it introduces the import feature's inherent issues in CDK, like:
+
+* CloudFormation doesn't check if the configuration of the physical resource
+  being imported matches the CloudFormation configuration of the corresponding
+  logical resource. Such import succeeds and leads to a drifted stack (to some
+  extent, the solution does validations on CDK level to minimise the risk).
+* Multiple resources can be imported in one IMPORT change set, but no other
+  changes are allowed - resources cannot be created or modified. This is
+  tricky to achieve with CDK's higher level constructs that produce multiple
+  CFN resources of which only some might need to be imported. The solution
+  will remove all new resources that are not to be imported from the template,
+  so the import operation can succeed, however, the deployed stack will not
+  be up to date with the CDK configuration after the import operation, another
+  `deploy` is needed.
+
+### What is the high-level project plan?
+
+A simple [PoC Draft PR](https://github.com/aws/aws-cdk/pull/17666) was created
+in the main `aws-cdk` repository that is being iteratively expanded with the
+proposed features. Once the approach is agreed and the agreed solution is
+implemented, the PR will be moved from "Draft" to "Open" status and CDK team
+member code review will be requested. Once all approved, the PR will be ready
+to be merged.
+
+### Are there any open issues that need to be addressed later?
+
+Not known at the time of writing
+
+## Appendix
+
+### Possible future expansions
+
+This proposal prioritizes implementation simplicity over user comfort
+(deliberately). For example, if a user needs to import a VPC, they will be
+prompted to type in a VPC ID, a value that is AWS-generated and must be first
+retrieved from Console/CLI. It would be much nicer to retrieve the list of VPCs
+in the target account and display to the user (including the Name tag) to choose
+from.
+
+Moreover, it would be of a great benefit to do pre-import validations in CDK
+(Does the physical resource exist? Does it not belong to another CloudFormation
+stack? Is the resource configuration consistent with the actual state?).
+
+However, such features would be difficult to implement and might require
+resource type-specific logic. Using [Cloud Control API](https://aws.amazon.com/blogs/aws/announcing-aws-cloud-control-api/)
+might avoid using service-specific APIs, but some resource type specific code
+might still be needed.
+
+It might be considered to release such convenience features iteratively
+per resource type, beginning with the most frequently used types.
```

Comment:
I think mass refactoring support (across multiple stacks) would be a great extension we could write down here.

It's going to require some additional tooling I think.

Reply:
done